mailpoet/lib/API/MP/v1/Segments.php

```diff
@@ -42,7 +42,9 @@ public function addList(array $data): array {
     $this->validateSegmentName($data);
 
     try {
-      $segment = $this->segmentsRepository->createOrUpdate($data['name'], $data['description'] ?? '');
+      $name = sanitize_text_field($data['name']);
```

Comment:
`sanitize_text_field` is just a wrapper around `strip_tags` which might not do what you might expect. For example:
```
> return [sanitize_text_field('hola < aa> "'), sanitize_text_field('hola > aa "')]
array (
  0 => 'hola < aa> "',
  1 => 'hola > aa "',
)
```
If this behavior is fine, please feel free to ignore this comment :)

Reply:
We still need to add esc_attr when printing the value on the output, so I guess this is fine cc @samnajian.